src/internal/components/dropdown/index.tsx

```diff
@@ -333,28 +336,46 @@ const Dropdown = ({
       <div className={clsx(stretchTriggerHeight && styles['stretch-trigger-height'])} ref={triggerRef}>
         {trigger}
       </div>
+
+      <TabTrap
+        focusNextCallback={() => dropdownRef.current && getFirstFocusable(dropdownRef.current)?.focus()}
+        disabled={!open || !trapFocus}
+      />
```

Comment:
Why is this trap needed? Why does this code not use the `<FocusLock>` wrapper?

Reply:
Thanks for asking :)

Ideally I'd love to wrap the dropdown in a focus-lock, like:

```
<FocusLock>
  <Trigger />
  <DropdownContainer>...</DropdownContainer>
</FocusLock>
```

However, this only works when expandToViewport=false. Otherwise, the dropdown container is rendered inside portal and the focus-lock utilities can't find the corresponding focus targets.

If you have some ideas on how to support that from the focus-lock I am happy to try it!

Reply:
This can be solvable by adopting `shards` prop from the original focus-lock: https://github.com/theKashey/react-focus-lock#portals

Fine for me to keep it this way for now

Reply:
Thanks, I mentioned the use case in the existing BRF ticket for generic focus-lock (18931).